docs/explanation/holistic-vs-delta-charms.md

```diff
@@ -1,66 +1,252 @@
+    # Provision a `Bar` resource in the workload
+    self.workload.create_bar(event.some_field)
 ```
 
+Notice how the handler methods map to Juju and custom events directly.
+There's less boilerplate and flow control is predictable.
+
+### Developing your charm
+
+The charm code and tests can get messy when event interdependencies are accounted for:
+
+- creating a resource is only possible after the workload has been installed
+- the available data typically needs to be mixed with application's configuration
+- cases where something is inferred from the sequence of events (see {external+juju:ref}`Juju | Hook ordering <hook>`)
+
+A good rule of thumb is this: if you're starting to use `defer` in various places, consider whether it's time to rewrite the charm using the reconciler pattern.
+
+## Which approach to use?
+
+At a high level, simple workloads are served well by delta charms.
+Complex workloads are more robust if the reconciler pattern is followed.
+The reconciler pattern is especially suitable for mature, feature-rich charms that use several charm libraries.
+
+### Dedicated handlers for specific Juju events
+
+Some events are special, and are typically processed outside the reconciler pattern in holistic charms.
+In both styles of charms a dedicated event handler is used for:
 
-## When to use the holistic approach
+- unit lifecycle end events, because the goal of reconciliation is different
+- action event, as it is synchronous and the event object holds arguments and results
+- secret rotation event, where secret must be modified synchronously
+- secret remove event, because the revision that needs removal is not available otherwise
+- Ops lifecycle events, like `on_collect_unit_status`, which accompany the Juju event
+- some custom events, if a specific charm library API is not suited for reconciliation
 
-If a charm is waiting for a collection of events, as in the example above, it makes sense to group those events together and handle them holistically, with a single code path.
+A good rule of thumb is this: if an event [cannot be deferred](#how-and-when-to-defer-events), it needs a dedicated handler.
 
-In other words, when writing a charm, it's not so much "should the *charm* be holistic?" as "does it make sense for *these events* to be handled holistically?"
+### Events observed by the reconcile method
 
-Using the holistic approach is normally centred around configuring an application. Various events that affect configuration use a common handler, to simplify writing an application config file and restarting the application.  This is common for events like `config-changed`, `relation-changed`, `secret-changed`, and `pebble-ready`.
+A holistic charm attaches the same observer to a group of events.
+In a charm following the reconciler pattern, the `_reconcile` method is attached to every interesting Juju event:
 
-Many existing charms use holistic event handling. A few examples are:
+- unit lifecycle start events
```

Comment:
The word "start" here helps, but similar to above, can you be more specific here?